src/config/backends/raspberry-pi.ts

```diff
@@ -0,0 +1,157 @@
+import * as _ from 'lodash';
+import { fs } from 'mz';
+
+import {
+	ConfigOptions,
+	DeviceConfigBackend,
+	bootMountPoint,
+	remountAndWriteAtomic,
+} from './backend';
+import * as constants from '../../lib/constants';
+import log from '../../lib/supervisor-console';
+
+/**
+ * A backend to handle Raspberry Pi host configuration
+ *
+ * Supports:
+ * 	- {BALENA|RESIN}_HOST_CONFIG_dtparam = value | "value" | "value1","value2"
+ * 	- {BALENA|RESIN}_HOST_CONFIG_dtoverlay = value | "value" | "value1","value2"
+ * 	- {BALENA|RESIN}_HOST_CONFIG_device_tree_param = value | "value" | "value1","value2"
+ * 	- {BALENA|RESIN}_HOST_CONFIG_device_tree_overlay = value | "value" | "value1","value2"
+ * 	- {BALENA|RESIN}_HOST_CONFIG_gpio = value | "value" | "value1","value2"
+ */
+
+export class RPiConfigBackend extends DeviceConfigBackend {
+	private static bootConfigVarPrefix = `${constants.hostConfigVarPrefix}CONFIG_`;
+	private static bootConfigPath = `${bootMountPoint}/config.txt`;
+
+	public static bootConfigVarRegex = new RegExp(
+		'(' + _.escapeRegExp(RPiConfigBackend.bootConfigVarPrefix) + ')(.+)',
+	);
+
+	private static arrayConfigKeys = [
+		'dtparam',
+		'dtoverlay',
+		'device_tree_param',
+		'device_tree_overlay',
+		'gpio',
+	];
+	private static forbiddenConfigKeys = [
+		'disable_commandline_tags',
+		'cmdline',
+		'kernel',
+		'kernel_address',
+		'kernel_old',
+		'ramfsfile',
+		'ramfsaddr',
+		'initramfs',
+		'device_tree_address',
+		'init_emmc_clock',
+		'avoid_safe_mode',
+	];
+
+	public matches(deviceType: string): boolean {
+		return _.startsWith(deviceType, 'raspberry') || deviceType === 'fincm3';
+	}
+
+	public async getBootConfig(): Promise<ConfigOptions> {
+		let configContents = '';
+
+		if (await fs.exists(RPiConfigBackend.bootConfigPath)) {
+			configContents = await fs.readFile(
+				RPiConfigBackend.bootConfigPath,
+				'utf-8',
+			);
+		} else {
+			await fs.writeFile(RPiConfigBackend.bootConfigPath, '');
+		}
+
+		const conf: ConfigOptions = {};
+		const configStatements = configContents.split(/\r?\n/);
+
+		for (const configStr of configStatements) {
+			// Don't show warnings for comments and empty lines
+			const trimmed = _.trimStart(configStr);
+			if (_.startsWith(trimmed, '#') || trimmed === '') {
+				continue;
+			}
+			let keyValue = /^([^=]+)=(.*)$/.exec(configStr);
+			if (keyValue != null) {
+				const [, key, value] = keyValue;
+				if (!_.includes(RPiConfigBackend.arrayConfigKeys, key)) {
+					conf[key] = value;
+				} else {
+					if (conf[key] == null) {
+						conf[key] = [];
+					}
+					const confArr = conf[key];
+					if (!_.isArray(confArr)) {
```

Comment:
```suggestion
					if (!Array.isArray(confArr)) {
```

Reply:
I've replaced all Lodash instance of Array checking with standard lib's version